src/core/qgsvectordataprovider.h

```diff
@@ -458,6 +458,12 @@ class CORE_EXPORT QgsVectorDataProvider : public QgsDataProvider, public QgsFeat
 
     /**
      * It returns false by default.
+     * Must be implemented by providers that support saving and loading metadata to db returning true
+     */
+    virtual bool isSaveAndLoadMetadataToDatabaseSupported() const;
```

Comment:
Instead of introducing this new method, maybe this should just be a new capability flag instead?

Reply:
Looks a good idea. There are already `isSaveAndLoadStyleToDatabaseSupported`  `isDeleteStyleFromDatabaseSupported`. Maybe I will focus first the metadata one and then come back just after on that. Is it ok?